tools/jdk/DumpPlatformClassPath.java

```diff
@@ -102,15 +102,21 @@ static boolean dumpJDK9AndNewerBootClassPath(
     // initialized bootclasspath data back out.
 
     Context context = new Context();
-    JavacTool.create()
+    try {
+      JavacTool.create()
         .getTask(
-            /* out = */ null,
-            /* fileManager = */ null,
-            /* diagnosticListener = */ null,
-            /* options = */ Arrays.asList("--system", String.valueOf(targetJavabase)),
-            /* classes = */ null,
-            /* compilationUnits = */ null,
-            context);
+          /* out = */ null,
+          /* fileManager = */ null,
+          /* diagnosticListener = */ null,
+          /* options = */ Arrays.asList("--system", String.valueOf(targetJavabase)),
+          /* classes = */ null,
+          /* compilationUnits = */ null,
+          context);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(String.format("Failed to collect system class path. Please ensure that the " +
+        "configured Java runtime ('%s') is a complete JDK. There are known issues with Homebrew versions of the " +
```

Comment:
How essential do you think the homebrew part of this is? It seems a little unfortunate to unconditionally mention homebrew for all users who see this, but if homebrew is the most common cause of this maybe it's the best option

Reply:
Judging by #15452 and my own experience, Homebrew is the only cause I know of. I mentioned it since Homebrew users may not realize that it didn't install a "complete JDK" for them. I could of course show this only on macOS or not at all and instead describe the required files in the `JAVA_HOME`.